### PR TITLE
refactor: consolidate heading classes and use tailwind layers

### PR DESCRIPTION
**Pull request by crnkovic**
## Summary

https://app.clickup.com/t/2u58u58

## Checklist

-   [ ] I checked my UI changes against the design and there are no notable differences
-   [ ] I checked my UI changes for any responsiveness issues
-   [ ] I checked my (code) changes for obvious issues, debug statements and commented code
-   [ ] I provided a screenshot of my changes to the component _(if applicable)_
-   [ ] I regenerated the `icons.html` file and checked if my newly added icon is shown correctly _(if necessary)_
-   [ ] I added an explanation on how to use the component to the readme _(if necessary)_
-   [ ] Documentation _(if necessary)_
-   [ ] Tests _(if necessary)_
-   [ ] Ready to be merged

### CODE REVIEW

File: resources/assets/css/_general.css [hunk 1]
```diff
@@ -9,62 +9,34 @@
...
 /* Navbar container */
 .navbar-container {
-    @apply px-8;
-}
-
-@screen md {
-    .navbar-container {
-        @apply px-10;
-    }
+    @apply px-8 md:px-10;
```
Comment (crnkovic) [inline resources/assets/css/_general.css:12]: replaced the `@screen` with `md:px-10`

File: resources/assets/css/_general.css [hunk 2]
```diff
@@ -9,62 +9,34 @@
...
-h1 {
-    @apply font-bold;
-}
-
-.dark h1,
-.dark h2,
-.dark h3,
-.dark h4 {
-    @apply text-theme-secondary-200;
+    @apply font-bold text-theme-secondary-900 dark:text-theme-secondary-200 leading-tight;
```
Comment (crnkovic) [inline resources/assets/css/_general.css:20]: additional `font-bold` on h1 was unnecessary. also `.dark h1` can be replaced with `dark:text-theme...`

File: resources/assets/css/_general.css [hunk 3]
```diff
@@ -9,62 +9,34 @@
...
-.dark header-4 {
-    @apply text-theme-secondary-200;
+    @apply font-bold text-theme-secondary-900 dark:text-theme-secondary-200 leading-tight;
 }
 
 /* Heading consistency */
 h1,
 .header-1 {
-    @apply text-3xl mb-1;
+    @apply text-3xl mb-1 md:text-4xl md:mb-2;
```
Comment (crnkovic) [inline resources/assets/css/_general.css:34]: replaced `@screen` with this

File: resources/assets/css/_general.css [hunk 4]
```diff
@@ -74,72 +46,37 @@ h3,
...
-    @apply text-lg mb-1;
+    @apply text-lg mb-1 md:text-2xl;
 }
 
-/* Header Responsiveness */
-@screen md {
-    h1,
-    .header-1 {
-        @apply text-4xl mb-2;
+@layer base {
```
Comment (crnkovic) [inline resources/assets/css/_general.css:52]: pushed core styles to `base` layer

File: resources/assets/css/_general.css [hunk 5]
```diff
@@ -74,72 +46,37 @@ h3,
...
-}
-
-.rotate-90 {
-    transform: rotate(90deg);
-}
-
-.rotate-180 {
-    transform: rotate(180deg);
-}
+@layer utilities {
```
Comment (crnkovic) [inline resources/assets/css/_general.css:71]: pushed `clear-transform` and `rotate-minus-90` to `utilities` layer. though they may not be even used in any of the projects, i wouldn't remove them here now

**Review by ItsANameToo — APPROVED** [review]

**Pull request merged**